src/Symfony/Component/Ldap/Adapter/AbstractConnection.php

```diff
@@ -24,22 +24,40 @@
     public function __construct(array $config = array())
     {
         $resolver = new OptionsResolver();
+
+        $this->configureOptions($resolver);
+
+        $this->config = $resolver->resolve($config);
+    }
+
+    /**
+     * Configures the adapter's options.
+     *
+     * @param OptionsResolver $resolver An OptionsResolver instance
+     */
+    protected function configureOptions(OptionsResolver $resolver)
+    {
         $resolver->setDefaults(array(
-            'host' => null,
-            'port' => 389,
+            'host' => 'localhost',
             'version' => 3,
-            'useSsl' => false,
-            'useStartTls' => false,
-            'optReferrals' => false,
+            'connection_string' => null,
+            'encryption' => 'none',
```

Comment:
`encryption` having a value of `none`, `tls` or `ssl`, I don't think something like `array('secure' => 'ssl')` would make it misleading, as we are only talking about encryption over the wire, not neither authentication or RBAC.